### DISCUSSION TITLE
Can Kargo perform argocd application update in a remote cluster.

### DISCUSSION THREAD

**Discussion by smileisak** [akuity/kargo#2705] [General]
I’m exploring the use of Kargo to manage stage promotions of Argo CD applications across our environments. In our setup, we have separate Argo CD instances deployed on each cluster for dev, UAT, and prod environments. Each Argo CD instance is isolated and manages deployments within its respective cluster.

Given that Kargo is designed to interact with Argo CD for streamlining application promotions, I wanted to ask:

- Does Kargo currently support managing stages across multiple, isolated Argo CD instances on separate clusters?
- If not, is there any plan on the roadmap to include this functionality in future releases of Kargo?

Any insights or recommendations on how to achieve stage management in this context using Kargo would be greatly appreciated. Thank you!

**Comment by krancour** [comment]
It does, but it does not work the way most people would suppose.

For both security and scalability reasons, we opted not to have Kargo controllers interact with multiple Argo CD control planes.

Instead, Kargo controllers should be distributed side-by-side with Argo CD control planes. Each will take to one Argo CD control plane and also "phone home" to a common Kargo control plane.

@fykaa will be working on documenting this.

**Reply by krancour** [reply]
This is a super rough draft of a diagram you may find helpful.

![image-2](https://github.com/user-attachments/assets/58a075a8-357b-4fbe-8468-341b396528bb)

**Comment by smileisak** [comment]
@krancour any updates on this ? Documentations or guides ?
